### PR TITLE
Write out titles JSON if --titlesJSONFile is passed to noninteractive-alignment-panel.py

### LINKED ISSUE #628: Add a --titleJSONFile option to noninteractive-alignment-panel.py
This will allow you to write out information about matching titles, initially each with a list of their bit scores.

### PR DESCRIPTION
**Pull request by terrycojones**
Fixes #628.

### CODE REVIEW

File: dark/titles.py [hunk 1]
```diff
@@ -35,6 +35,21 @@ def __init__(self, read, hsps):
...
+        """
+        Get information about a title alignment as a dictionary.
+
+        @return: A C{dict} representation of the title aligment.
+        """
+        return {
+            'hsps': [hsp.toDict() for hsp in self.hsps],
+            'read': {
+                'id': self.read.id,
+                'quality': self.read.quality,
```
Comment (bamueh) [inline dark/titles.py]: This is going to fail if it's not a fastq read, or not?

File: dark/titles.py [hunk 2]
```diff
@@ -482,3 +510,15 @@ def tabSeparatedSummary(self, sortOn=None):
                 '%(subjectTitle)s',
             ]) % titleSummary)
         return '\n'.join(result)
+
+    def toDict(self):
+        """
+        Get information about the titles alignments as a dictionary.
```
Comment (bamueh) [inline dark/titles.py]: s/titles alignments/TitlesAlignments

File: dark/titles.py [hunk 3]
```diff
@@ -482,3 +510,15 @@ def tabSeparatedSummary(self, sortOn=None):
...
+
+        @return: A C{dict} representation of the titles aligments.
```
Comment (bamueh) [inline dark/titles.py]: Same as above

File: dark/titles.py (hunk 1 above)
Comment (bamueh) [reply dark/titles.py]: Could you put a toDict method on Read() or something?

Comment (terrycojones) [reply dark/titles.py]: No, the JSON will just have a `null` if the quality is `None`.

Comment (terrycojones) [reply dark/titles.py]: Thanks - the Read class already had that method, and now I'm using it :-)

File: dark/titles.py (hunk 2 above)
Comment (terrycojones) [reply dark/titles.py]: I try to use regular English in the description of the methods and put the actual classes in `C{...}` in the `@param` sections.

File: dark/titles.py (hunk 3 above)
Comment (terrycojones) [reply dark/titles.py]: Same as above, too :-)

**Pull request merged**